src/mjolnir/bssbuilder.cc

```diff
@@ -197,7 +197,16 @@ DirectedEdge make_directed_edge(const GraphId endnode,
                                 const uint32_t oppo_local_idx) {
   DirectedEdge directededge;
   directededge.set_endnode(endnode);
-  directededge.set_length(std::max(1.0f, valhalla::midgard::length(shape)));
+  float length = valhalla::midgard::length(shape);
+
+  if (length > kMaxEdgeLength) {
+    // Consider this a catastrophic error.  Need to do the check here as
+    // transit edges could be large; however, they should not be for the graph.
+    LOG_ERROR("Exceeding max. edge length: " + std::to_string(length));
+    throw std::runtime_error("DirectedEdgeBuilder: exceeded maximum edge length");
+  }
```

Comment:
can we move this `if` block into `directededge::set_length` and replace the warning that is in there? then it is always going to happen in one place and we wont have to duplicate it whenever we make edges in some part of the code.

Reply:
@kevinkreiser the reason I didn't do this was because of transit and that we may have large edges.  Unless you think the max is too large even for transit.

Reply:
updated to use bool in set_length